### PR TITLE
Auto-update & publish in Github action

### PR DESCRIPTION
**Pull request by yannlugrin**
This PR is made to remove all Dockerfile and script files from the repo. We generate the files on the fly to build and test the image. When PR is merged in dev, the image is also pushed to the Docker repository.

The `update.sh` and `test.sh` script are also merged in a `tools.sh` script and used to generate Dockerfile and script files then run tests. Docker buildx tool is used to build the image and push it to the repository.

### CODE REVIEW

**Comment by yannlugrin** [comment]
Hey @WengerK and @gido,

This is a suggestion to remove all Dockerfile and script files from the repository; plus publish the image to docker hub with Docker action instead of Codeship.

**Comment by yannlugrin** [comment]
FYI we can also push the image to the Github repository (both are possible at the same time).

File: .github/workflows/container-structure-test.yml [hunk 1]
```diff
@@ -12,13 +14,38 @@ jobs:
...
+      - name: Generate docker image config files for PHP ${{ matrix.php_version }}
+        run: ./tools.sh --generate=${{ matrix.php_version }}
+      - name: Build docker image for PHP ${{ matrix.php_version }}
+        uses: docker/build-push-action@v2
+        with:
+          context: ./php/${{ matrix.php_version }}
+          file: ./php/${{ matrix.php_version }}/Dockerfile
+          tags: antistatique/php-dev:${{ matrix.php_version }}
+          cache-to: type=local,dest=/tmp/.buildx-cache
+          load: true
```
Comment (gido) [inline .github/workflows/container-structure-test.yml:37]: can we explicitly set `push: false` ?

File: .github/workflows/container-structure-test.yml [hunk 2]
```diff
@@ -12,13 +14,38 @@ jobs:
...
-      - name: Build docker images PHP Only
-        run: docker build -t antistatique/php-dev:${{ matrix.php_version }} -f ./php/${{ matrix.php_version }}/Dockerfile .
-      - name: Run tests PHP only
-        run: container-structure-test test --image antistatique/php-dev:${{ matrix.php_version }} --config ./tests/baseMetadataTests.yaml --config ./tests/baseFileExistenceTests.yaml --config ./php/${{ matrix.php_version }}/tests/config--${{ matrix.php_version }}.yaml
-
+      - name: Login to DockerHub
+        uses: docker/login-action@v1
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
```
Comment (gido) [inline .github/workflows/container-structure-test.yml:27]: can we move the login step before the push step or this is a requirement for `docker-build-push-actions` ?

File: tools.sh [hunk 3]
```diff
@@ -16,27 +18,44 @@ while [ $# -gt 0 ]; do
        echo " "
        echo "options:"
        echo "-h, --help                show brief help"
+       echo "-g, --generate=VERSION    VERSION is optional, all images config files are generated by default; set a VERSION like '7.2-node8'"
        echo "-b, --build=VERSION       VERSION is optional, all images are build by default; set a VERSION like '7.2-node8'"
+       echo "-t, --test=VERSION        VERSION is optional, all images are tested by default; set a VERSION like '7.2-node8'"
        echo "--publish=VERSION         set a VERSION like '7.2-node8' or 'all'; publish also build images"
```
Comment (gido) [inline tools.sh:24]: can we remove the `--publish` option ?

**Review by gido — APPROVED** [review]

**Comment by gido** [comment]
We need to remove the codeship files and update the README

File: tools.sh (hunk 3 above)
Comment (gido) [reply tools.sh]: done in ca1298ad86ee45a5bef73539bc0c68df2c42e8b0

File: .github/workflows/container-structure-test.yml (hunk 2 above)
Comment (gido) [reply .github/workflows/container-structure-test.yml]: ok make sense to have it here, as the build pull images from docker hub.

**Comment by gido** [comment]
> We need to remove the codeship files and update the README

Done.

**Pull request merged**